server/routes/genesetEnrichment.ts

```diff
@@ -0,0 +1,120 @@
+import { genesetEnrichmentRequest, genesetEnrichmentResponse } from '../shared/types/routes/genesetEnrichment.ts'
+import fs from 'fs'
+import { spawn } from 'child_process'
+import { Readable } from 'stream'
+import path from 'path'
+import serverconfig from '../src/serverconfig.js'
+
+export const api = {
+	endpoint: 'genesetEnrichment',
+	methods: {
+		all: {
+			init,
+			request: {
+				typeId: 'genesetEnrichmentRequest'
+			},
+			response: {
+				typeId: 'genesetEnrichmentResponse'
+				// will combine this with type checker
+				//valid: (t) => {}
+			}
+		}
+	}
+}
+
+function init({ genomes }) {
+	return async (req: any, res: any): Promise<void> => {
+		try {
+			const results = await run_genesetEnrichment_analysis(req.query as genesetEnrichmentRequest, genomes)
+			res.send(results as genesetEnrichmentResponse)
+		} catch (e: any) {
+			res.send({ status: 'error', error: e.message || e })
+		}
+	}
+}
+
+async function run_genesetEnrichment_analysis(q: genesetEnrichmentRequest, genomes: any) {
+	if (!genomes[q.genome].termdbs) throw 'termdb database is not available for ' + q.genome
+
+	const genesetenrichment_input = {
+		genes: q.genes,
+		fold_change: q.fold_change,
+		db: genomes[q.genome].termdbs.msigdb.cohort.db.connection.name,
```

Comment:
please leave a comment here:

as initial prototype it's okay to hardcode `msigdb`. but it's intended to be a request parameter coming from client, as `genome.termdbs{}` is designed to support other geneset db besides msigdb